File: Deepspace_ruby/Deepspace/lib/EnemyStarShip.rb
#encoding:utf-8
module Deepspace
class EnemyStarShip
  
  attr_reader :ammoPower, :name, :shieldPower, :loot, :damage
  
  def initialize(n,a,s,l,d)
    @ammoPower = a
    @name = n
    @shieldPower = s
    @loot = l
    @damage = d
  end
  
  def self.newCopy(copy)
    EnemyStarShip.new(copy.name, copy.ammoPower, copy.shieldPower, copy.loot, copy.damage)
  end
  
  def getUIversion()
    EnemyToUI.new(self)
  end
  
  def protection
    @shieldPower
  end
  
  def fire
    @ammoPower
  end
  
  def receiveShot(shot)
    if shot > @shieldPower
      return ShotResult::DONOTRESIST
    else
      return ShotResult::RESIST
    end
  end
  
  def to_s
      return "\nName: #{@name} AmmoPower: #{@ammoPower} ShieldPower: #{@shieldPower} Loot:#{@loot} Damage:#{@damage}"
  end
  
  
end
end

File: Deepspace_java/Deepspace/src/deepspace/EnemyStarShip.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author victor
 */
public class EnemyStarShip {
    private float ammoPower, shieldPower;
    private String name;
    private Loot loot;
    private Damage damage;

    EnemyStarShip(String name, float ammoPower, float shieldPower,Loot loot, Damage damage) {
        this.ammoPower = ammoPower;
        this.shieldPower = shieldPower;
        this.name = name;
        this.loot = loot;
        this.damage = damage;
    }
    
    EnemyStarShip(EnemyStarShip e) {
        this.ammoPower = e.ammoPower;
        this.shieldPower = e.shieldPower;
        this.name = e.name;
        this.loot = e.loot;
        this.damage = e.damage;
        
    }

    public float getAmmoPower() {
        return ammoPower;
    }

    public float getShieldPower() {
        return shieldPower;
    }

    public String getName() {
        return name;
    }

    public Loot getLoot() {
        return loot;
    }

    public Damage getDamage() {
        return damage;
    }
    
    EnemyToUI getUIversion() {
        return new EnemyToUI(this);
    }
    
    public float fire() {
       return ammoPower;
    }
    
    public float protection() {
        return shieldPower;
    }
    
    public ShotResult receiveShot(float shot) {
        if(shot >= shieldPower)
            return ShotResult.DONOTRESIST;
        else
            return ShotResult.RESIST;            
    }

    @Override
    public String toString() {
        return "EnemyStarShip{" + "ammoPower=" + ammoPower + ", shieldPower=" + shieldPower + ", name=" + name + ", loot=" + loot + ", damage=" + damage + '}';
    }
    
    
    
    
    
    
    
}
